logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/user/UploadUserAvatarUseCase.kt

```diff
@@ -14,21 +12,20 @@ interface UploadUserAvatarUseCase {
      * Function allowing the upload of a user profile picture (avatar)
      * This first will upload the data as an asset and then will link this asset with the [User]
      *
-     * @param mimeType mimetype of the user picture
-     * @param imageData binary data of the actual picture
+     * @param imageData binary data of the actual jpeg picture
      * @return UploadAvatarResult with [UserAssetId] in case of success or [CoreFailure] on failure
      */
-    suspend operator fun invoke(mimeType: String, imageData: ByteArray): UploadAvatarResult
+    suspend operator fun invoke(imageData: ByteArray): UploadAvatarResult
 }
 
 internal class UploadUserAvatarUseCaseImpl(
     private val userDataSource: UserRepository,
     private val assetDataSource: AssetRepository
 ) : UploadUserAvatarUseCase {
 
-    override suspend operator fun invoke(mimeType: String, imageData: ByteArray): UploadAvatarResult = suspending {
-        assetDataSource
-            .uploadAndPersistPublicAsset(UploadAssetData(imageData, ImageAsset.JPG, true, RetentionType.ETERNAL))
+    override suspend operator fun invoke(imageData: ByteArray): UploadAvatarResult = suspending {
+        val mimeType = ImageAsset.JPEG
```

Comment:
I think the idea was to basically use JPEG as the defined format for compression and storage optimisation reasons.

Reply:
Even if it's compressed, the Type should not be hard codded

Reply:
Does that mean that we want to allow users to upload raw image formats or not so lossy ones like .raw or .png?

Reply:
Let's just stick to this JPEG hardcoded version of the mimetype for now and revisit it in case there are any future issues or we decide to limit the avatar image size.